deploy_aws.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+# This script installs and configures the required command-line utilities, creates an EC2 instance with the specified AMI ID and security group, and prints the public IP address of the instance.
+
+# Homebrew should be installed
+if ( which brew > /dev/null ) 
+then
+  echo -e "==== homebrew already installed ===="
+else 
+  NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+  echo "==== installing homebrew ===="
+fi
+
+# Install aws cli
+if ( which aws > /dev/null ) 
+then
+  echo -e "\n==== awscli already installed ===="
+else 
+  brew install awscli
+  echo "\n==== installing awscli ===="
+fi
+
+# Install jq
+if ( which jq > /dev/null ) 
+then
+  echo -e "\n==== jq already installed ===="
+else 
+  brew install jq
+  echo "\n==== installing jq ===="
+fi
+
+# Check if an aws key-pair is present
+#  pipe the key pair to jq and write that key pair to a file named id_rsa 
+if ( aws ec2 describe-key-pairs --key-name aws_rsa_key_pair --out text )
+then
+  echo -e "==== key pair present ===="
+else
+  echo -e "==== creating key pair ===="
+  aws ec2 create-key-pair --key-name aws_rsa_key_pair --query 'KeyMaterial' --output text > aws_rsa_key_pair.pem 
+  chmod 0600 aws_rsa_key_pair.pem 
+fi
+
+# Check if the security group exists (if it is -z zero length)
+security_group_id=$(aws ec2 describe-security-groups --group-names ssh_http --region $AWS_DEFAULT_REGION --output json | jq -r '.SecurityGroups[0].GroupId')
```

Comment:
Maybe make this one functional and remove variable like the others?

Reply:
https://github.com/Jason-Doze/aws_deploy/pull/2/commits/841aef5655b7830be5b486b0d6cafd567cbdf2c4